### DISCUSSION TITLE
Is it possible to subscribe to 2 or more AWS SQS queue with-in  the same application

### DISCUSSION THREAD

**Discussion by madhub** [aws/aws-dotnet-messaging#124] [Q&A] [answered]
For example  if I have 2 SQS queue's https://sqs.us-east-1.amazonaws.com/012345678910/MyQueue1, https://sqs.us-east-1.amazonaws.com/012345678910/MyQueue2, it is possible to subscribe & process messages from **MyQueue1** & **MyQueue2** with in the same application ?

By looking at the library , it looks like it is possible by adding **AddSQSPoller** with different queue names. My understanding is that if a **message** is registered with **handler** , irrespective from **which queue message is received**,  if handler is found for a **message** it will be invoked. In other wards **handlers** are not **tied to the queue** . Is this correct ?
```csharp
builder.Services.AddAWSMessageBus(builder =>
{
    builder.AddSQSPoller("https://sqs.us-east-1.amazonaws.com/012345678910/MyQueue1");
    builder.AddMessageHandler<ChatMessageHandler, ChatMessage>("chatMessage");

    builder.AddSQSPoller("https://sqs.us-east-1.amazonaws.com/012345678910/SomeOtherQueue");
    builder.AddMessageHandler<ChatMessageHandler, ChatMessage>("chatMessage");
});
```

**Comment by ashovlin** [comment, marked as answer]
Yes, you can subscribe to multiple queues from a polling application. A single [`BackgroundService`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.backgroundservice) will poll both queues, with a long-running task for each queue.

You only need to register the message handler one time, they are not tied to a specific queue. Note that they are registered and retrieved as [scoped](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection#scoped) services, so the handler will be created once per each message that is received.
* https://github.com/awslabs/aws-dotnet-messaging/blob/db7703ff5eb4a17c7450ee55713b41c7cac2fc4a/src/AWS.Messaging/Services/HandlerInvoker.cs#L57-L59
